Mathlib/NumberTheory/MulChar/Basic.lean

```diff
@@ -467,7 +476,23 @@ def ringHomComp (χ : MulChar R R') (f : R' →+* R'') : MulChar R R'' :=
     map_nonunit' := fun a ha => by simp only [map_nonunit χ ha, map_zero] }
 #align mul_char.ring_hom_comp MulChar.ringHomComp
 
+lemma injective_ringHomComp {f : R' →+* R''} (hf : Function.Injective f) :
+    Function.Injective (ringHomComp (R := R) · f) := by
+  simpa only [Function.Injective, ext_iff, ringHomComp, coe_mk, MonoidHom.coe_mk, OneHom.coe_mk]
+    using fun χ χ' h a ↦ hf (h a)
+
+lemma ringHomComp_eq_one_iff {f : R' →+* R''} (hf : Function.Injective f) {χ : MulChar R R'} :
+    χ.ringHomComp f = 1 ↔ χ = 1 := by
+  conv_lhs => rw [← (show  (1 : MulChar R R').ringHomComp f = 1 by ext; simp)]
+  exact (injective_ringHomComp hf).eq_iff
+
+lemma ringHomComp_ne_one_iff {f : R' →+* R''} (hf : Function.Injective f) {χ : MulChar R R'} :
+    χ.ringHomComp f ≠ 1 ↔ χ ≠ 1 := by
+  simp only [Ne, ringHomComp_eq_one_iff, hf]
```

Comment:
```suggestion
lemma ringHomComp_ne_one_iff {f : R' →+* R''} (hf : Function.Injective f) {χ : MulChar R R'} :
    χ.ringHomComp f ≠ 1 ↔ χ ≠ 1 :=
  (ringHomComp_eq_one_iff hf).not
```
Is it common to have both equivalences in Mathlib?

Reply:
I was following Yaël's treatment of `AddChar` which also has both. If we're using this in rewrites, it will be annoying to unfold the definition of `≠`.

Reply:
I agree it's more convenient for rewriting (although one could use `rw [(ringHomComp_eq_one_iff hf).not]`).